op-e2e/alt-sync_test.go

```diff
@@ -0,0 +1,338 @@
+		published = append(published, payload.ExecutionPayload.ID())
+	}
+	// Blocks are now received via the RPC based alt-sync method
+	cfg.Nodes[RoleSeq].Tracer = seqTracer
+
+	sys, err := cfg.Start(t)
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	l2Seq := sys.NodeClient(RoleSeq)
+
+	// Transactor Account
+	ethPrivKey := cfg.Secrets.Alice
+
+	// Submit a TX to L2 sequencer node
+	receiptSeq := SendL2Tx(t, cfg, l2Seq, ethPrivKey, func(opts *TxOpts) {
+		opts.ToAddr = &common.Address{0xff, 0xff}
+		opts.Value = big.NewInt(1_000_000_000)
+	})
+	t.Logf("tx receipt is in block %v", receiptSeq.BlockNumber)
+
+	assert.EqualValues(t, cfg.DeployConfig.L2BlockTime, 1)
+
+	// Gossip has been limited to a single block, so now we wait until the block for which we hold
+	// the receipt is no longer available in the sequencer's outbound gossip window.
+	targetPublishedLen := receiptSeq.BlockNumber.Int64() + sequencerOutboundQueueSize + 1
+	for {
+		blocksPublished := int64(len(getPublished()))
+		if blocksPublished >= targetPublishedLen {
+			break
+		}
+		t.Logf("waiting for blocks published (%v) >= %v", blocksPublished, targetPublishedLen)
+		time.Sleep(time.Second)
+	}
+	// Give time for the outbound gossip queues to lose messages.
+	time.Sleep(time.Second)
+	t.Logf("starting syncers after %v blocks published", len(getPublished()))
+
+	var syncers []*syncerType
+
+	sequencerNode := sys.RollupNodes[RoleSeq]
+
+	addSyncer := func(i int) {
+		name := fmt.Sprintf("syncer-%d", i)
+		newSyncer := makeSyncer(ctx, t, name, cfg, sys)
+		// Link to all the other syncers
+		for _, syncer := range syncers {
+			linkAndConnectNodeNamesNodes(t, sys.Mocknet, syncer.peerId, newSyncer.peerId)
+		}
+		// And to the sequencer.
+		linkAndConnectNodeNamesNodes(t, sys.Mocknet, sequencerNode.P2P().Host().ID(), newSyncer.peerId)
+		syncers = append(syncers, newSyncer)
+	}
+
+	// Approximately mirroring the replica count in prod.
+	for i := range iter.N(15) {
+		addSyncer(i)
+	}
+
+	eg, ctx := errgroup.WithContext(ctx)
+
+	for _, syncer := range syncers {
+		// Behold, Go 1.21.
+		syncer := syncer
+		eg.Go(func() error {
+			syncer.requireAltSyncTx(ctx, t, receiptSeq, getPublished)
+			t.Logf("%v synced", syncer.name)
+			return nil
+		})
+	}
+	require.NoError(t, eg.Wait())
+
+	// Don't stop the nodes right away so they can sync from each other.
+	for _, syncer := range syncers {
+		syncer.stop()
+	}
+
+	altSyncSources := make(map[peer.ID]int)
+	for _, syncer := range syncers {
+		fmt.Printf("%v (%v)\n", syncer.peerId, syncer.name)
+		for source, blocks := range syncer.syncedPayloads {
+			fmt.Printf("  %v (%v):", source, len(blocks))
+			for _, block := range blocks {
+				fmt.Printf(" %v", block.blockId.Number)
+			}
+			fmt.Printf("\n")
+		}
+		for _, block := range syncer.syncedPayloads[p2p.PayloadSourceAltSync] {
+			altSyncSources[block.from]++
+		}
+	}
+	//peerIds := slices.SortedFunc(maps.Keys(altSyncSources), func(id peer.ID, id2 peer.ID) int {
```

Comment:
dead code?

Reply:
Ah, this was the code for Go 1.23... Unfortunately no good to us yet.